### PR TITLE
Issues/538

### LINKED ISSUE #538: Provide summary fact for JBoss Fuse
Summarize JBoss Fuse information as yes/maybe/no decision.

### LINKED ISSUE #539: Provide summary fact for BRMS
Summarize BRMS information as yes/maybe/no decision.

### PR DESCRIPTION
**Pull request by noahl**
Closes #538 , closes #539 .

### CODE REVIEW

File: rho/facts.py [hunk 1]
```diff
@@ -202,6 +204,8 @@ def new_fact(fact_name, description, is_default=None,
 new_fact('jboss.fuse-on-karaf.find-karaf-jar',
          'Use the find utility to locate karaf.jar',
          is_default=False)
+new_fact('jboss.fuse.summary', 'Summary of Fuse info',
```
Comment (chambridge) [inline rho/facts.py]: Let's be consistent and say `Summary info of Fuse information`

File: rho/facts.py [hunk 2]
```diff
@@ -193,6 +193,8 @@ def new_fact(fact_name, description, is_default=None,
 new_fact('jboss.eap.summary', 'Summary of JBoss EAP information',
          is_default=True, categories=[JBOSS_FACTS])
 new_fact('jboss.activemq-ver', 'ActiveMQ version', is_default=False)
+new_fact('jboss.brms.summary', 'Summary of BRMS information',
```
Comment (chambridge) [inline rho/facts.py:196]: Should this be in the JBOSS_FACTS category? like EAP and Fuse?

File: rho/postprocessing.py [hunk 3]
```diff
@@ -813,21 +821,31 @@ def process_fuse_on_eap(fact_names, host_vars):
...
     """Process karaf_home indicators to detect Fuse-on-Karaf."""
 
-    if JBOSS_FUSE_ON_KARAF_KARAF_HOME not in fact_names:
+    log.info('In process_karaf_home')
+
+    if (JBOSS_FUSE_ON_KARAF_KARAF_HOME not in fact_names and
+            JBOSS_FUSE_SUMMARY not in fact_names):  # noqa
         return {}
 
+    log.info('Past first return')
```
Comment (chambridge) [inline rho/postprocessing.py]: Is this logging valuable?

File: rho/postprocessing.py [hunk 4]
```diff
@@ -813,21 +821,31 @@ def process_fuse_on_eap(fact_names, host_vars):
...
+
     if 'karaf_homes' not in host_vars:
         return {JBOSS_FUSE_ON_KARAF_KARAF_HOME:
                 'Error: fact karaf_homes not collected.'}
     karaf_homes = host_vars['karaf_homes']
 
+    log.info('Past second return')
```
Comment (chambridge) [inline rho/postprocessing.py]: Is this logging valuable?

File: rho/postprocessing.py [hunk 5]
```diff
@@ -813,21 +821,31 @@ def process_fuse_on_eap(fact_names, host_vars):
...
+
     err, bin_fuse = raw_output_present(fact_names, host_vars,
                                        JBOSS_FUSE_ON_KARAF_KARAF_HOME,
                                        'karaf_home_bin_fuse',
                                        'ls -1 KARAF_HOME/bin/fuse')
+    log.info('Raw output: %s, %s', err, bin_fuse)
     if err is not None:
         return err
 
+    log.info('Past third return')
```
Comment (chambridge) [inline rho/postprocessing.py]: Is this logging valuable?

File: rho/postprocessing.py [hunk 6]
```diff
@@ -836,24 +854,35 @@ def process_karaf_home(fact_names, host_vars):
     if err is not None:
         return err
 
-    system_org_jboss_results, _ = process_indicator_files(['fuse'],
-                                                          system_org_jboss)
+    log.info('Past fourth return')
```
Comment (chambridge) [inline rho/postprocessing.py]: Is this logging valuable?

File: rho/facts.py (hunk 2 above)
Comment (noahl) [reply rho/facts.py:196]: Yep, thanks for catching that.

File: rho/facts.py (hunk 1 above)
Comment (noahl) [reply rho/facts.py]: Done

File: rho/postprocessing.py (hunk 3 above)
Comment (noahl) [reply rho/postprocessing.py]: Sorry, I left a bunch of debug logging in. I took it all out.

File: rho/postprocessing.py (hunk 4 above)
Comment (noahl) [reply rho/postprocessing.py]: Gone.

File: rho/postprocessing.py (hunk 5 above)
Comment (noahl) [reply rho/postprocessing.py]: Gone.

File: rho/postprocessing.py (hunk 6 above)
Comment (noahl) [reply rho/postprocessing.py]: Gone.

**Comment by noahl** [comment]
@chambridge I think I've fixed everything you pointed out, please take another look!

**Review by chambridge — APPROVED** [review]

**Pull request merged**